src/fileseq/filesequence.py

```diff
@@ -221,7 +242,70 @@ def setPadding(self, padding):
             padding (str): sequence padding to set
         """
         self._pad = padding
-        self._zfill = self.__class__.getPaddingNum(self._pad)
+        self._frame_pad, _, self._subframe_pad = self._pad.partition('.')
+        self._zfill = self.getPaddingNum(self._frame_pad)
+        self._decimal_places = self.getPaddingNum(self._subframe_pad)
+
+    def usdPadding(self):
+        """
+        Return the the padding characters in the sequence in USD format.
+
+        Returns:
+            str: sequence padding
+        """
+        pad = self.getUsdPaddingChars(self._zfill)
+        if self._decimal_places:
+            pad = ".".join([pad, self.getUsdPaddingChars(self._decimal_places)])
+        return pad
+
+    def framePadding(self):
+        """
+        Return the the padding characters in the sequence.
+
+        Returns:
+            str: sequence padding
+        """
+        return self._frame_pad
+
+    def setFramePadding(self, padding):
+        """
+        Set new padding characters for the frames of the sequence.
+        i.e. "#" or "@@@" or '%04d', or an empty string to disable range formatting.
+
+        Args:
+            padding (str): sequence padding to set
+        """
+        self._frame_pad = padding
+        if self._subframe_pad:
+            self._pad = '.'.join([self._frame_pad, self._subframe_pad])
+        else:
+            self._pad = self._frame_pad
+        self._zfill = self.getPaddingNum(self._frame_pad)
+
+    def subframePadding(self):
+        """
+        Return the the padding characters for subframes in the sequence.
+
+        Returns:
+            str: sequence padding
+        """
+        return self._subframe_pad
+
+    def setSubframePadding(self, padding):
```

Comment:
Technically `FileSequence` could have the original simpler methods without subframe handling in many places, and `SubFileSequence` could exclusively handle subframe logic, but it would either involve a lot of duplication, or function decomposition solely for the purpose of separating the common logic. To me that seems like it would make everything more difficult to understand, so I just went for a subclass with different regular expressions that engaged the subframe logic.

We could move the subframe-specific functions (such as `setSubframePadding`) to `SubFileSequence`, but will also need to update logic to avoid calling these functions or updating subframe members from `FileSequence` (e.g. `FileSequence.__init__` shouldn't set `self._subframe_pad`).

Thoughts?

Reply:
I do see what you mean about it being a bit more complicated to break apart a lot of the logic so that `SubFileSequence` can have a hook back into those spots for the purpose of the subframe code. On the otherhand it does feel a bit strange that `FileSequence` partially supports subframe features but you need a subclass in order to opt into the actual parsing. 
Would it be weird to make them protected no-ops on `FileSequence` and then move the implementation to `SubFileSequence` and add public accessors? That would at least make the public interface clear that you should use the subclass to get full subframe support, while internally the `FileSequence` class can still call the protected methods. Its just a wild though and I am open to your opinion on it. 
Do you think it would be important for `FileSequence` and `SubFileSequence` to have a 1-to-1 public API so that you can treat instances transparently? If that is the goal then we would need to keep those public accessors on the super class.